USAGOV-1557 cf logs to new relic

arpage opened the pull request:
<!--- Provide a general summary of your changes in the title above -->
## Jira Task

<!--- Provide a link to the Jira ticket -->
https://cm-jira.usa.gov/browse/USAGOV-1557

## Description
<!--- Summarize the changes made in this pull request, not what it's for. -->

## Type of Changes
<!--- Put an `x` in all the boxes that apply. -->
- [ ] New Feature
- [ ] Bugfix
- [ ] Frontend (Twig, Sass, JS)
  - Add screenshot showing what it should look like
- [ ] Drupal Config (requires "drush cim")
- [ ] New Modules (requires rebuild)
- [ ] Documentation
- [ ] Infrastructure
  - [ ] CMS
  - [ ] WAF
  - [ ] Egress
  - [ ] Tools
- [ ] Other

## Testing Instructions
<!-- This instructions are different from “testing instructions” in Jira – those are typically for Content/UX stakeholders -->
<!-- Not “see Jira” – if they are really the same, copy and paste. -->

### Change Requirements
<!-- Checkboxes to indicate need for changes to some part of the system -->

- [ ] Requires New Documentation (Link: {})
- [ ] Requires New Config
- [ ] Requires New Content

### Validation Steps

- Test instruction 1
- Test instruction 2
- Test instruction 3

## Security Review
<!-- Checkboxes to indicate need for review -->

- [ ] Adds/updates software (including a library or Drupal module)
- [ ] Communication with external service
- [ ] Changes permissions or workflow
- [ ] Requires SSPP updates


## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.

arpage committed:
USAGOV-1557-cf-logs-to-new-relic-unofficial-merges: Stricter cf space setting, helper scripts

arpage committed:
USAGOV-1557-cf-logs-to-new-relic-unofficial-merges: Merge remote-tracking branch 'origin/dev' into USAGOV-1557-cf-logs-to-new-relic-unofficial-merges

arpage committed:
USAGOV-1557-cf-logs-to-new-relic: Deployment scripts

arpage committed:
USAGOV-1557-cf-logs-to-new-relic: Merge remote-tracking branch 'origin/dev' into USAGOV-1557-cf-logs-to-new-relic

arpage committed:
USAGOV-1557-cf-logs-to-new-relic: Remove hardcoded cf space names.

arpage committed:
USAGOV-1557-cf-logs-to-new-relic: Add basic howto for cf task apps

arpage committed:
Merge branch 'dev' into USAGOV-1557-cf-logs-to-new-relic

arpage committed:
USAGOV-1557-cf-logs-to-new-relic: Merge remote-tracking branch 'origin/dev' into USAGOV-1557-cf-logs-to-new-relic

arpage committed:
USAGOV-1557-cf-logs-to-new-relic-alpine: Fix S3 cli path issue, revert container to use alpine as a base, fix date cmd syntax differences between alpine (busybox-based date cmd) and ubuntu (gnu date).

akf committed:
Merge branch 'dev' into USAGOV-1557-cf-logs-to-new-relic

akf commented on scripts/cfevents/space-guids:
```diff
@@ -0,0 +1,5 @@
+export dev=ea450105-895e-4a33-9541-6a0df35ee626
```
oof, and should there be a `dr` here as well?

akf replied:
Actually, it would probably be worth using `cf space dev --guid` to gather these guids programatically.

arpage replied:
file removed.   Single space GUID is now gathered in the script where it is needed.

akf reviewed (commented):
I haven't tested this, but it looks reasonable. It has reminded me that I wanted to ask about a couple of things, which maybe we should just talk through: 

1. You're doing some setup in .profile and then you run `source ~/.profile` in the scripts for the actual tasks. Jacob and I found this a little confusing, as he started from the assumption that .profile had done the setup and didn't realize that he needed to copy that in elsewhere. Since .profile doesn't automatically run for the tasks, I think it would be less confusing to put this setup in a differently-named "include" file. I could be wrong. 
2. cf roles! I would like the event capturer to run as a service user with OrgAuditor and SpaceAuditor roles, and no Manager or Developer roles, if possible. I realize that the task runner probably needs SpaceDeveloper in order to run tasks.

arpage committed:
USAGOV-1557-cf-logs-to-new-relic-alpine: Fixes brokenness.

arpage committed:
USAGOV-1557-cf-logs-to-new-relic-alpine: Merge remote-tracking branch 'origin/dev' into USAGOV-1557-cf-logs-to-new-relic-alpine

arpage committed:
USAGOV-1557-cf-logs-to-new-relic: Merge remote-tracking branch 'origin/dev' into USAGOV-1557-cf-logs-to-new-relic

arpage committed:
USAGOV-1557-cf-logs-to-new-relic: Merge branch 'USAGOV-1557-cf-logs-to-new-relic-alpine' into USAGOV-1557-cf-logs-to-new-relic

arpage committed:
USAGOV-1557-cf-logs-to-new-relic: Remove test files

arpage commented:
> 1. You're doing some setup in .profile and then you run `source ~/.profile` in the scripts for the actual tasks. Jacob and I found this a little confusing, as he started from the assumption that .profile had done the setup and didn't realize that he needed to copy that in elsewhere. Since .profile doesn't automatically run for the tasks, I think it would be less confusing to put this setup in a differently-named "include" file. I could be wrong.

The profile isn't read unless cf ssh (or a shell invoked with cf ssh) is run with the "login shell" option.   So, it needs to be sourced manually.   I like the include idea, however it is sort of the same thing - we should discuss.

> 2. cf roles! I would like the event capturer to run as a service user with OrgAuditor and SpaceAuditor roles, and no Manager or Developer roles, if possible. I realize that the task runner probably needs SpaceDeveloper in order to run tasks.

I agree, and will investigate.

arpage committed:
Merge branch 'dev' into USAGOV-1557-cf-logs-to-new-relic

arpage committed:
USAGOV-1557-cf-logs-to-new-relic: Remove space developer role, added Space and Org Auditor roles.

arpage committed:
USAGOV-1557-cf-logs-to-new-relic: Add coreutils and tzdata packages for GNU time formatting capabilities, replace time contants w/ variables defined in .profile

arpage commented:
> > 2. cf roles! I would like the event capturer to run as a service user with OrgAuditor and SpaceAuditor roles, and no Manager or Developer roles, if possible. I realize that the task runner probably needs SpaceDeveloper in order to run tasks.
> 
> I agree, and will investigate.

I've removed space developer and added org/space auditor.  Everything seems to run OK.

arpage committed:
USAGOV-1557-cf-logs-to-new-relic: Fix dockerhub auth for cf push

arpage commented:
Added fix for dockerhub auth during cf push

arpage committed:
Merge branch 'dev' into USAGOV-1557-cf-logs-to-new-relic

arpage committed:
USAGOV-1557-cf-logs-to-new-relic: Reduce memory to 128MB

arpage committed:
USAGOV-1557-cf-logs-to-new-relic: Allow cfevents deployment via circleci

arpage committed:
Merge branch 'dev' into USAGOV-1557-cf-logs-to-new-relic

arpage committed:
USAGOV-1557-cf-logs-to-new-relic: Update manifest for memory/disk defaults
